torch_xla/distributed/fsdp/xla_flatten_params_wrapper.py

```diff
@@ -440,14 +444,15 @@ def unflatten_params(self,
       if orig_flattened:
         self._flatten_params(orig_flat_params)
 
-  def __getattr__(self, name: str) -> Any:
+  def __getattr__(self, name: str) -> Union[Tensor, nn.Module]:
     """Forward missing attributes to wrapped module."""
     try:
       return super().__getattr__(name)  # defer to nn.Module's logic
     except AttributeError:
       return getattr(self.module, name)  # fallback to wrapped module
-
-  def __getitem__(self, key: int) -> Any:
+      
+  T = TypeVar('T', bound=Module)
+  def __getitem__(self, key: int) -> Union[nn.Sequential, T]:
```

Comment:
I think one would need to add `from typing import TypeVar` in this file?

Also, this method is just to return the `nn.Module` instances in case the base module is a `nn.Sequential`, so I think we can just use `def __getitem__(self, key: int) -> nn.Module:` as the signature here.

Reply:
Changed